src/main/java/cam72cam/immersiverailroading/gui/overlay/GuiBuilder.java

```diff
@@ -74,79 +74,86 @@ protected GuiBuilder(JsonObject data) throws IOException {
         }
 
         // Text stuff
-        if (data.has("text")) {
-            JsonObject txt = data.get("text").getAsJsonObject();
-            text = txt.get("value").getAsString();
-            textHeight = txt.has("height") ? txt.get("height").getAsFloat() : 8;
+        DataBlock txt = data.getBlock("text");
+        if (txt != null) {
+            text = txt.getValue("value").asString();
+            textHeight = txt.getValue("height").asFloat(0f);
         } else {
             text = null;
             textHeight = 0;
         }
 
         // Controls
-        this.readout = data.has("readout") ? Readouts.valueOf(data.get("readout").getAsString().toUpperCase(Locale.ROOT)) : null;
-        this.control = data.has("control") ? data.get("control").getAsString() : null;
-        this.invert = data.has("invert") && data.get("invert").getAsBoolean();
-        this.hide = data.has("hide") && data.get("hide").getAsBoolean();
-
-        if (data.has("translate")) {
-            JsonObject tl = data.get("translate").getAsJsonObject();
-            this.tlx = tl.has("x") ? tl.get("x").getAsFloat() : 0;
-            this.tly = tl.has("y") ? tl.get("y").getAsFloat() : 0;
+        String readout = data.getValue("readout").asString();
+        this.readout = readout != null ? Readouts.valueOf(readout.toUpperCase(Locale.ROOT)) : null;
```

Comment:
could this cause issues on different JVMs? Shouldn't there just be an "official" IR Locale used for data exchange between the mod and the models?

Reply:
I'm not super familiar with java locales, I just defaulted to this as it seemed like a sane default recommended by others.